config/config.go

```diff
@@ -0,0 +1,237 @@
+// config.go - mixnet client configuration
+// Copyright (C) 2017  David Anthony Stainton
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package provides mixnet client configuration utilities
+package config
+
+import (
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/katzenpost/client/constants"
+	"github.com/katzenpost/client/crypto/vault"
+	"github.com/katzenpost/core/crypto/ecdh"
+	"github.com/katzenpost/core/crypto/rand"
+	"github.com/op/go-logging"
+	"github.com/pelletier/go-toml"
+)
+
+var log = logging.MustGetLogger("mixclient")
+
+// Account is used to deserialize the account sections
+// of the configuration file
+type Account struct {
+	Name     string
+	Provider string
+}
+
+// ProviderPinning is used to deserialize the
+// provider pinning sections of the configuration file
+type ProviderPinning struct {
+	PublicKeyFile string
+	Name          string
+}
+
+// SMTPProxy is used to deserialize the
+// smtp section of the configuration file
+type SMTPProxy struct {
+	Address string
+	Network string
+}
+
+// POP3Proxy is used to deserialize the pop3
+// section of the configuration file
+type POP3Proxy struct {
+	Address string
+	Network string
+}
+
+// Config is used to deserialize the configuration file
+type Config struct {
+	Account         []Account
+	ProviderPinning []ProviderPinning
+	SMTPProxy       SMTPProxy
+	POP3Proxy       POP3Proxy
```

Comment:
Indeed, I agree; however I am using golang's JSON unmarshal feature as described here:
https://golang.org/pkg/encoding/json/#Unmarshal

where the struct elements correspond to the named elements in the JSON text. It's the easy way to do it... but probably there's another way.

Reply:
Since field names of SMTPProxy and POP3Proxy are the same, you could have a single type 'Proxy' with the net and address fields, and have this be
```
    SMTPProxy    Proxy
    POP3Proxy    Proxy
```

Reply:
yes that sounds good. i'll do that.